lib/std/zig/system/macos.zig

```diff
@@ -7,452 +7,382 @@ const std = @import("std");
 const assert = std.debug.assert;
 const mem = std.mem;
 
-pub fn version_from_build(build: []const u8) !std.builtin.Version {
-    // build format:
-    //   19E287 (example)
-    //   xxyzzz
+/// Detect macOS version.
+/// On error `os` will not be modified.
+pub fn detect(os: *std.Target.Os) !void {
+    // Drop use of osproductversion sysctl because:
+    //   1. only available 10.13.4 High Sierra and later
+    //   2. when used from a binary built against < SDK 11.0 it returns 10.16 and masks Big Sur 11.x version
     //
-    // major = 10
-    // minor = x - 4 = 19 - 4 = 15
-    // patch = ascii(y) - 'A' = 'E' - 'A' = 69 - 65 = 4
-    // answer: 10.15.4
+    // New method is to parse and use the first valid:
+    //   /System/Library/CoreServices/.SystemVersionPlatform.plist
+    //   /System/Library/CoreServices/SystemVersion.plist
```

Comment:
These are available regardless of the fact you have Xcode or CLT installed, right?

Reply:
Correct, comes with vanilla macOS, no xcode or command line tools and that is absolute file name, not offset from SDK or anything. An interesting tidbit is that the xnu-kernel itself, at boot/load learns the value for what it thinks os version is from that primary file.

`. SystemVersionPlatform.plist` is **new** and may be a Big Sur and higher thing. I don't see it on 10.13.6 High Sierra system for example. I'm confident they put that in as the only way to skirt context-sensitive version presentation.